File: pset1/mario/C/mario_hard.c
#include <stdio.h>

int main(void) {
    // Get height
    int height;
    do {
        printf("Height: ");
        scanf("%d", &height);
    }
    while (height < 1 || height > 8);

    // Each row
    for (int i = 0; i < height; i ++) {
        int spaces = height - i -1;
        int hashes = i + 1;

        // Left side
        for (int j = 0; j < spaces; j ++) {
            printf(" ");
        }

        for (int j = 0; j < hashes; j ++) {
            printf("#");
        }

        // Middle
        printf("  ");
        
        // Right side
        for (int j = 0; j < hashes; j ++) {
            printf("#");
        }
        printf("\n");
    }
}

File: pset1/mario/python/mario_hard.py
# Get height
height = 0
while height < 1 or height > 8:
    height = int(input("Height: "))

for i in range(height):
    spaces = height - i - 1
    hashes = i + 1

    # Left side
    for j in range(spaces):
        print(" ", end = "")
    
    for j in range(hashes):
        print("#", end = "")
    
    # Middle
    print("  ", end = "")

    # Right side
    for j in range(hashes):
        print("#", end = "")
    print("")